packages/analytics/amplify_analytics_pinpoint/lib/src/legacy_native_data_provider/flutter_legacy_native_data_provider_io.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:io' show Platform;
+
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.android.dart';
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.ios.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+class FlutterLegacyNativeDataProvider implements LegacyNativeDataProvider {
+  /// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+  factory FlutterLegacyNativeDataProvider() {
+    final provider = Platform.isIOS ? DataProviderIos() : DataProviderAndroid();
```

Comment:
question: Maybe I'm missing something here, but is this using `DataProviderAndroid` for all non-ios io platforms (desktop windows, macos, linux)? When I ran example app on macos I got this error:


```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(channel-error, Unable to establish connection on channel., null, null)
#0      PigeonDataProvider.initialize (package:amplify_analytics_pinpoint/src/legacy_native_data_provider/pigeon_data_provider.android.g.dart:40:7)
<asynchronous suspension>
#1      AmplifyAnalyticsPinpointDart.configure (package:amplify_analytics_pinpoint_dart/src/analytics_plugin_impl.dart:169:7)
<asynchronous suspension>
#2      Future.wait.<anonymous closure> (dart:async/future.dart:522:21)
<asynchronous suspension>
#3      AmplifyClassImpl.configurePlatform (package:amplify_core/src/amplify_class_impl.dart:73:5)
<asynchronous suspension>
#4      AmplifyClass.configure (package:amplify_core/src/amplify_class.dart:107:7)
<asynchronous suspension>
#5      _MyAppState._configureAmplify (package:amplify_analytics_pinpoint_example/main.dart:70:7)
<asynchronous suspension>
```

Reply:
Thanks good catch, I'll need to fix this then